js/classesAndBaseFunction.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+let playOrRecord;
+let activePlayer;
+let singlyOrTogetherPlay;
+const COORDINATETOP = 200;
+const COORDINATELEFTPleyer = 700;
+const COORDINATELEFTPComputer = 305;
+const WIDTH = 30;
+const fields = [[], []];
+const Ships = [[], []];
+
+class Ship {
+  constructor(leng, num, player) {
+    this.length = leng;
+    this.coordinate = [];
+    this.number = num;
+    this.who = player;
+    this.killed = [];
+    this.full = false;
+  }
+
+  drawShip() {
+    for (let i = 0; i < this.coordinate.length; i++) {
+      const cord = this.coordinate[i];
+      drawRectangle(cord[0], cord[1], this.who, 'blue');
+    }
+  }
+  drawKilled() {
+    for (let i = 0; i < this.killed.length; i++) {
+      const kill = this.killed[i];
+      drawRectangle(kill[0], kill[1], this.who, 'red');
+    }
+  }
+
+}
+
+class Fields {
+  constructor(player, num) {
+    this.sea = [];
+    for (let i = 0; i < 10; i++) {
+      this.sea.push([]);
+      for (let j = 0; j < 10; j++) {
+        this.sea[i][j] = num;
+      }
+    }
+    this.who = player;
+
+  }
+  clean() {
+    for (let i = 0; i < 10; i++) {
+      for (let j = 0; j < 10; j++) {
+        this.sea[i][j] = 0;
+      }
+    }
+  }
+  drawMask() {
+    for (let i = 0; i < 10; i++) {
+      for (let j = 0; j < 10; j++) {
+        if (this.sea[i][j] === -1) drawPoint(i, j, this.who, 'black');
+
+      }
+    }
+
+  }
+}
+
+const hiddenArray = array => {
+  for (let i = 0; i < array.length; i++) {
+    if (document.getElementById(array[i]).hidden === true) {
+      document.getElementById(array[i]).hidden = false;
+    } else document.getElementById(array[i]).hidden = true;
+  }
+};
+
+const cleanShips = player => {
+  Ships[player].splice(0, 10);
+};
+
+const coordinate = (x, y) => {
+  let coordinateX;
+  let coordinateY;
+  let coordinateLeft;
+  if (activePlayer)  coordinateLeft = COORDINATELEFTPleyer;
+  else coordinateLeft = COORDINATELEFTPComputer;
+  if (y > COORDINATETOP && y < COORDINATETOP + WIDTH * 10) {
+    coordinateY = Math.floor((y - COORDINATETOP) / WIDTH);
+    if (x > coordinateLeft && x < coordinateLeft + WIDTH * 10) {
+      coordinateX = Math.floor((x - coordinateLeft) / WIDTH);
+    } else setTimeout(WhereClick, 10);
+    if (playOrRecord) {
+      checkSinglyShip(coordinateY, coordinateX);
+      return true;
+
+    } else {
+      checkBedingPlayer(coordinateY, coordinateX);
+      return true;
+    }
+  } else {
+    setTimeout(WhereClick, 10);
+    return false;
+  }
+};
+
+function CLICK(e) {
+  const pageX = e.pageX;
+  const pageY = e.pageY;
+  coordinate(pageX, pageY);
+  window.removeEventListener('click', CLICK, false);
+}
+
+const WhereClick = () => {
+  window.addEventListener('click', CLICK, false);
+  return 0;
+};
+
+const recordAroundCell = (i, j, field, num, boolean, number, cell) => {
```

Comment:
Very weird function, based on the number of iterations and initial values this could probably be just a few `if`s.